gcloud-java-compute/src/test/java/com/google/cloud/compute/OperationTest.java

```diff
@@ -353,6 +365,109 @@ public void testIsDone_NotExists() throws Exception {
+    assertNull(operation.waitFor());
+    verify(mockOptions);
+  }
+
+  @Test
+  public void testWaitForCheckingPeriod() throws InterruptedException, TimeoutException {
+    initializeExpectedOperation(5);
+    Compute.OperationOption[] expectedOptions =
+        {Compute.OperationOption.fields(Compute.OperationField.STATUS)};
+    TimeUnit timeUnit = createStrictMock(TimeUnit.class);
+    timeUnit.sleep(42);
+    EasyMock.expectLastCall();
+    Operation runningOperation = Operation.fromPb(serviceMockReturnsOptions,
+        globalOperation.toPb().setError(null).setStatus("RUNNING"));
+    Operation completedOperation =
+        Operation.fromPb(serviceMockReturnsOptions, globalOperation.toPb().setError(null));
+    expect(compute.options()).andReturn(mockOptions);
+    expect(mockOptions.clock()).andReturn(Clock.defaultClock());
+    expect(compute.getOperation(GLOBAL_OPERATION_ID, expectedOptions)).andReturn(runningOperation);
+    expect(compute.getOperation(GLOBAL_OPERATION_ID, expectedOptions))
+        .andReturn(completedOperation);
+    expect(compute.getOperation(GLOBAL_OPERATION_ID)).andReturn(completedOperation);
+    replay(compute, timeUnit, mockOptions);
+    initializeOperation();
+    assertSame(completedOperation, operation.waitFor(WaitForOption.checkEvery(42, timeUnit)));
+    verify(timeUnit, mockOptions);
+  }
+
+  @Test
+  public void testWaitForCheckingPeriod_Null() throws InterruptedException, TimeoutException {
+    initializeExpectedOperation(4);
+    Compute.OperationOption[] expectedOptions =
+        {Compute.OperationOption.fields(Compute.OperationField.STATUS)};
+    TimeUnit timeUnit = createStrictMock(TimeUnit.class);
+    timeUnit.sleep(42);
+    EasyMock.expectLastCall();
+    Operation runningOperation = Operation.fromPb(serviceMockReturnsOptions,
+        globalOperation.toPb().setError(null).setStatus("RUNNING"));
+    expect(compute.options()).andReturn(mockOptions);
+    expect(mockOptions.clock()).andReturn(Clock.defaultClock());
+    expect(compute.getOperation(GLOBAL_OPERATION_ID, expectedOptions)).andReturn(runningOperation);
+    expect(compute.getOperation(GLOBAL_OPERATION_ID, expectedOptions)).andReturn(null);
+    expect(compute.getOperation(GLOBAL_OPERATION_ID)).andReturn(null);
+    replay(compute, timeUnit, mockOptions);
+    initializeOperation();
+    assertNull(operation.waitFor(WaitForOption.checkEvery(42, timeUnit)));
+    verify(compute, timeUnit, mockOptions);
+  }
+
+  @Test
+  public void testWaitForWithTimeout() throws InterruptedException, TimeoutException {
+    initializeExpectedOperation(4);
+    Compute.OperationOption[] expectedOptions =
+        {Compute.OperationOption.fields(Compute.OperationField.STATUS)};
+    TimeUnit timeUnit = createStrictMock(TimeUnit.class);
+    timeUnit.sleep(1);
+    EasyMock.expectLastCall();
+    Clock clock = createStrictMock(Clock.class);
+    EasyMock.expect(clock.millis()).andReturn(0L);
+    EasyMock.expect(clock.millis()).andReturn(1L);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'EasyMock.expect' due to existing static import 'org.easymock.EasyMock.expect'](https://www.codacy.com/app/mziccard/gcloud-java/file/2653889166/issues/source?bid=3123103&fileBranchId=3307743#l454)